appservice/src/intTest/groovy/nu/yona/server/UserTest.groovy

```diff
@@ -77,11 +78,87 @@ class UserTest extends AbstractAppServiceIntegrationTest
 		def response = appService.deleteUser(john)
 
 		then:
-		response.status == 200
+		appService.assertResponseStatus(response, 200)
 		def getUserResponse = appService.getUser(john.url, false)
 		getUserResponse.status == 400 || getUserResponse.status == 404
 	}
 
+	def 'Create John Doe with an Android device'()
+	{
+		given:
+		def ts = timestamp
+
+		when:
+		def johnAsCreated = createJohnDoe(ts, "My S8", "ANDROID")
+
+		then:
+		def johnAfterNumberConfirmation = appService.confirmMobileNumber(
+				{
+					AppService.assertResponseStatusSuccess(it)
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices".size == 1
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices"[0].name == "My S8"
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices"[0].operatingSystem == "ANDROID"
+				}, johnAsCreated)
+
+		cleanup:
+		appService.deleteUser(johnAsCreated)
+	}
+
+	def 'Create John Doe with an iOS device'()
```

Comment:
This test and the following 3 are specific for the devices feature. Wouldn't it be more logical to put them in a separate test class? The `UserTest` class is already big.

Reply:
True. I've put them here because this test has code to create an arbitrary user. It's easier to fit these tests in here then in the more "rigid" Richard and Bob structure. Once we implement the device maintenance and device change buddy messaging, we need a dedicated device test. I was thinking that we should move these tests there at that time.

Reply:
It was actually pretty simple, so I've moved them to DeviceTest.groovy